[BUG] Otto fail marshal to json a array of string to object

Otto version: c382bd3c16ff2fef9b5fe0dd8bf4c4ec6bfe62c1

Actual: Otto marshal a javascript array of string to a JSON object
Expected: I expected that Otto marshal a javascript array to a JSON array 

Output:
```
foo,bar
object
foo,bar
object
{"Bar":["foo","bar"],"Foo":{}}
panic: json: cannot unmarshal object into Go struct field Foobar.Foo of type []string

goroutine 1 [running]:
main.main()
        /tmp/pippo/main.go:59 +0x4a8
exit status 2
```

Example code:
```
package main

import (
	"encoding/json"
	"fmt"

	"github.com/robertkrimen/otto"
)

func main() {
	var err error
	type Foobar struct {
		Foo []string	
		Bar []string	

	}

	var value Foobar
	var temp map[string]interface{}

	//Decode value to temp
	tempRaw, err := json.Marshal(value)
	if err != nil {
		panic(err)
	}
	err = json.Unmarshal(tempRaw, &temp)
	if err != nil {
		panic(err)
	}

	//run otto
	vm := otto.New()
	err = vm.Set("foobar", temp) //using &value instead of temp work in this example but not in my exact usecase 
	if err != nil {
		panic(err)
	}
	err = vm.Set("bar", []string{"foo", "bar"})
	if err != nil {
		panic(err)
	}
	_, err = vm.Run(`
		console.log([ 'foo', 'bar' ]);
		console.log(typeof [ 'foo', 'bar' ]);
		console.log(bar);
		console.log(typeof bar);
		foobar.Foo = [ 'foo', 'bar' ]; //don't work
		foobar.Bar = bar; //work
	`)
	if err != nil {
		panic(err)
	}

	//Encode temp to value
	tempRaw, err = json.Marshal(temp)
	if err != nil {
		panic(err)
	}
	fmt.Println(string(tempRaw))
	err = json.Unmarshal(tempRaw, &value)
	if err != nil {
		panic(err)
	}

	fmt.Println(value)
}
```

--- a/issue_test.go
+++ b/issue_test.go
@@ -928,3 +928,18 @@ func Test_issue386(t *testing.T) {
 		})
 	}
 }
+
+func Test_issue382(t *testing.T) {
+	vm := New()
+	vm.Set("panicFunc", func(call FunctionCall) Value {
+		panic("test")
+	})
+	_, err := vm.Run(`
+		try {
+			panicFunc()
+		} catch (err) {
+			console.log("panic triggered:", err)
+		}
+	`)
+	require.NoError(t, err)
+}

--- a/runtime.go
+++ b/runtime.go
@@ -121,7 +121,7 @@ func (self *_runtime) tryCatchEvaluate(inner func() Value) (tryValue Value, exce
 	// throw = Something was thrown
 	// throwValue = The value of what was thrown
 	// other = Something that changes flow (return, break, continue) that is not a throw
-	// Otherwise, some sort of unknown panic happened, we'll just propagate it
+	// Otherwise, some sort of unknown panic happened, we'll just propagate it.
 	defer func() {
 		if caught := recover(); caught != nil {
 			if exception, ok := caught.(*_exception); ok {
@@ -135,13 +135,13 @@ func (self *_runtime) tryCatchEvaluate(inner func() Value) (tryValue Value, exce
 				exception = true
 				tryValue = caught
 			default:
-				panic(caught)
+				exception = true
+				tryValue = toValue(caught)
 			}
 		}
 	}()
 
-	tryValue = inner()
-	return
+	return inner(), false
 }
 
 // toObject

